### PR TITLE
8284077: Create an automated test for JDK-4170173

### PR DESCRIPTION
**Pull request by srmandal**
Create an automated test for [JDK-4170173](https://bugs.openjdk.java.net/browse/JDK-4170173)

Issue
JTextComponent.AccessibleJTextComponent.getAfterIndex(int part, int index) works incorrectly, when 'part' parameter is AccessibleText.WORD.
It returns a space (" ") instead of the correct word.

The test verifies the fix for this behavior by checking the getAfterIndex for AccessibleText.CHARACTER,AccessibleText.WORD,AccessibleText.SENTENCE for the components JTextField, JTextArea, JEditorPane.

While working on this test case there was a related bug relevant to this [JDK-4170174](https://bugs.openjdk.java.net/browse/JDK-4170174) 
This is marked as duplicate, addressess a similar issue.
It indicates that JTextComponent.AccessibleJTextComponent.getBeforeIndex(int part, int index) works incorrectly, when part parameter is AccessibleText.WORD.
It returns a space (" ") instead of correct word.

Hence an additional test was added for this, for verifying the behavior of getBeforeIndex.
AccessibleText.CHARACTER,AccessibleText.WORD,AccessibleText.SENTENCE for the components JTextField, JTextArea, JEditorPane.

The two tests have multiple distinct assertions. For this reason, as well as for maintainability, the two were not clubbed into a single test.
However, the two tests are still similar in the functional flow of the code and the functionality they are testing as well - hence they have been clubbed into a single review.
This review is for migrating tests from a closed test suite to open.

Testing:
The tests ran successfully on Mach5 with multiple runs (30) on windows-x64, linux-x64 and macos-x64.

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8284077](https://bugs.openjdk.java.net/browse/JDK-8284077): Create an automated test for JDK-4170173

### Reviewers
 * [Sergey Bylokhov](https://openjdk.java.net/census#serb) (@mrserb - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.java.net/jdk pull/8138/head:pull/8138` \
`$ git checkout pull/8138`

Update a local copy of the PR: \
`$ git checkout pull/8138` \
`$ git pull https://git.openjdk.java.net/jdk pull/8138/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 8138`

View PR using the GUI difftool: \
`$ git pr show -t 8138`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.java.net/jdk/pull/8138.diff">https://git.openjdk.java.net/jdk/pull/8138.diff</a>

</details>

### CODE REVIEW

File: test/jdk/javax/accessibility/4170173/AccessibleJTextAfterIndexTest.java [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+            new JEditorPane("text/plain", "Test1 Test2 Test3.\nTest4 Test5");
+    }
+
+    private static boolean checkAccessibleText(JTextComponent jTextComponent,
+        int character, int characterPosition, String expectedString)
+            throws Exception {
+        SwingUtilities.invokeAndWait(() -> actualAccessibleText =
+            jTextComponent.getAccessibleContext().getAccessibleText()
+            .getAfterIndex(character, characterPosition));
+
```
Comment (mrserb) [inline test/jdk/javax/accessibility/4170173/AccessibleJTextAfterIndexTest.java:114]: It feels that the code above is too smart=) Can we simplify/reformat it a little bit. I do not think that you need to jump from the main thread to the EDT and back for each testcase, just wrap the "doTest" by the invokeAndWait

Comment (srmandal) [reply test/jdk/javax/accessibility/4170173/AccessibleJTextAfterIndexTest.java]: This is a really good suggestion, thank you. I have simplified the test.

**Comment by srmandal** [comment]
@mrserb can you please review the changes after I have incorporated the earlier feedback ?

**Review by mrserb — APPROVED** [review]

**Comment by srmandal** [comment]
@mrserb Can you please sponsor this PR?

**Pull request closed without merging**